src/CLRunner.ts

```diff
@@ -335,6 +335,70 @@ export class CLRunner {
         return startResponse
     }
 
+    private async StartSession(appId: string, createParams: CLM.SessionCreateParams): Promise<CLM.Session> {
+
+        // Don't save logs on server if custom storage was provided
+        if (CLState.logStorage) {
+            createParams.saveToLog = false
+        }
+
+        const session = await this.clClient.StartSession(appId, createParams as CLM.SessionCreateParams)
+
+        // If using customer storage add to log storage
+        if (CLState.logStorage) {
+            // For self-hosted log storage logDialogId is hash of sessionId
+            session.logDialogId = CLM.hashText(session.logDialogId)
+            const logDialog: CLM.LogDialog = {
+                logDialogId: session.logDialogId,
+                packageId: session.packageId,
+                rounds: [],
+                initialFilledEntities: [],
+                targetTrainDialogIds: [],
+                createdDateTime: session.createdDatetime,
+                dialogBeginDatetime: new Date().getTime().toString(),
+                dialogEndDatetime: new Date().getTime().toString(),
```

Comment:
I assume this is just for initial PR but would remember to change the Begin , End times to be correct.

Reply:
at initialization they both start out the same.  End time is updated when dialog is edited